src/solvers/dg/2d/dg.jl

```diff
@@ -634,6 +649,49 @@ function init_mortar_connectivity!(elements, mortars, mesh::TreeMesh{2})
 end
 
 
+function init_boundary_conditions(n_boundaries_per_direction, mesh::TreeMesh{2})
+  # "eval is evil"
+  # This is a temporary hack until we have switched to a library based approach
+  # with pure Julia code instead of parameter files.
+  bcs = parameter("boundary_conditions", ["nothing", "nothing", "nothing", "nothing"])
+  if bcs isa AbstractArray
+    boundary_conditions = eval.(Symbol.(bcs))
+  else
+    # This adds support for using a scalar boundary condition (like 'periodicity = "false"')
+    boundary_conditions = eval.(Symbol.([bcs for _ in 1:n_directions(mesh.tree)]))
+  end
+
+  # Sanity checks about specifying boundary conditions
+  for direction in 1:n_directions(mesh.tree)
+    bc = boundary_conditions[direction]
+    count = n_boundaries_per_direction[direction]
+    if direction == 1
+      dir = "-x"
+    elseif direction == 2
+      dir = "+x"
+    elseif direction == 3
+      dir = "-y"
+    else
+      dir = "+y"
+    end
+
+    # Check #1: All directions with boundaries should have function boundary conditions
+    if count > 0 && !(bc isa Function)
+      error("Found $(count) boundaries in the $(dir)-direction, but corresponding boundary " *
+            "condition '$(get_name(bc))' is not a function")
+    end
+
+    # Check #2: All boundary conditions should be a function or "nothing"
+    if !(bc isa Function || bc === nothing)
+      error("Bad parameter: boundary condition '$(get_name(bc))' in the $(dir)-direction must " *
+            "be either a function or 'nothing'")
+    end
+  end
```

Comment:
The restriction to `::Function` is too narrow when we move to a more Julia approach, I think. We should also allow users to pass `struct`s that can be called.

Reply:
Should I define something like
```julia
iscallable(o) = !isempty(methods(o))
```
as suggested [here](https://discourse.julialang.org/t/how-to-determine-if-an-object-is-callable/2897/3)? It's not very Julian (as discussed in the referenced thread), but OTOH we only use it for user convenience (as I think it is more helpful than a cryptic error somewhere down the line when the BCs are first applied).

Reply:
But that still doesn't check whether a method with the correct signature is available.

Reply:
Yeah, but that cannot be checked except by calling it, right? If you feel like this is overkill or that it is too un-Julian, I can also remove this check.

Reply:
I would remove this additional check. It could only help in very few cases where users do something totally wrong. But if you want, you can keep a check like `!isempty(methods(o))` - I don't have a strong opinion here.

I just think it's more likely that 
- people want to pass in a function defined outside of Trixi (where this fails, cf. 9b949dcecca9b76d0a2c6ddc6cc808d37fb5b48c)
- people make a type somewhere in the signature

Reply:
OK, I took out check no. 2 in 5f28281 and now only make sure that the user did not forget to set anything (i.e., that it is not "nothing").